### ISSUE TITLE
Deep-Cover shows syntax error when parsing a file

### ISSUE THREAD

**Issue by abotalov** [deep-cover/deep-cover#44]
Consider this file:
```ruby
require 'deep-cover'
require 'pathname'

class A
  def c=(obj)
    puts obj
  end
end

def func(*b)
  a = A.new
  a&.c = *b
end
```
When `deep-cover clone ruby test.rb` is invoked, the following error is shown:

```
test.rb:12: syntax error, unexpected *
...ov[0][11]+=1;_temp=_temp&.c = *(_temp=[*b];$_cov[0][9]+=1;_t...
...                              ^
test.rb:12: syntax error, unexpected ';', expecting &. or :: or '[' or '.'
...b];$_cov[0][9]+=1;_temp=_temp);$_cov[0][10]+=1;;_temp;else;n...
```

**Comment by marcandre** [comment]
Thanks for the bug report 😸
I'm on it.

**Comment by marcandre** [comment]
Interesting issue. All comes down to the fact that `foo = bar = *baz` is not valid Ruby syntax (which I find surprising). Using `foo = (bar = *baz)` is ok though, so that's what I've changed it to.
Releasing when travis is green.

**Comment by marcandre** [comment]
Fixed in 0.7.1.
Thanks again for the bug report!
